### ISSUE TITLE
Show Battery Status on Battery Powered Devices

### ISSUE THREAD

**Issue by ghost** [nextcloud/serverinfo#290]
**Is your feature request related to a problem? Please describe.**
It seems that serverinfo repo is not active so I submit it here.
I host my NC on a battery powered device. I have to mind the battery level for fear that it may go poweroff.
However, to make it more durable, it's headless and the X11 is not enabled by default, which means I need another solution to check the battery status.

**Describe the solution you'd like**
- [x] Display the battery capacity by reading ``/sys/class/power_supply/*/capacity``.
- - [ ] Multi battery support? I have only mono battery devices so I cannot debug it.
- [x] Detect charging status by reading ``/sys/class/power_supply/*/status``.
- [x] BSD support
- - [ ] Detect charging type. Yet to come up with a way to implement.
- [x] Monitor the battery usage by displaying a chart. 
- - [x] Prompt admins to enable Cron.
- - [x] Cron background jobs.
- - [x] Migration and creation of the new table ``oc_serverinfo_battery``. 
- [x] API support.
- - [ ] What sort of useful data should the API respond?
- [x] Warn admin when battery is low. Currently warns at the level of 50%.
- - [ ] How low? 25%? 50%?
- - [x] Drop ``app.php`` and use new ``Application.php``. 
- [x] L10N support.
- [x] UI Improvements.
- - [x] Make the chart pretty. Could use the help from an expert on UI design...
- - [x] Currently using modified Gnome icons. Not sure if it's licensed.
- - [x] Add a live battery icon that changes with the battery capacity.

## Screenshot
![0](https://user-images.githubusercontent.com/75371007/113994645-25acfc00-9888-11eb-93b4-4d3e26ae1c7c.png)
![1](https://user-images.githubusercontent.com/75371007/113994654-2776bf80-9888-11eb-95e4-3916e6907788.png)
![2](https://user-images.githubusercontent.com/75371007/113994656-2776bf80-9888-11eb-9ffb-34bf2da85a8a.png)

**Comment by joshtrichards** [comment]
While this response is going to a @ghost account, I'll comment for the record.

At first I thought this was solely a detailed request and a compelling visual mock-up, but it looks like you may have gotten pretty far along in your implementation. Unfortunately I don't see an associated PR. I hope you found what you needed on your journey!

I sympathize with your need, but I don't really see why this needs to be integrated into NC itself. I suspect there are some reasonable solutions that don't require being integrated with NC at all. Alternatively, you might be able to move your implemented functionality into a dedicated NC app. Lastly, I can't find any evidence that anyone else has ever requested a battery related feature for NC. 

Tldr: IMO this falls outside the scope of NC/serverinfo. That said, I'm only one voice.

**Comment by kesselb** [comment]
As this sounds like a nice feature, the requests for this are quite low. Currently there are no plans to implement such a feature. Thus I will close this ticket for now. This does not mean we don't want this feature, but it is simply not on our roadmap for the near future. If somebody wants to implement this feature nevertheless we are happy to assist and help out.
